pkg/northbound/admin/admin.go

```diff
@@ -137,18 +137,17 @@ func (s Server) RollbackNetworkChange(
 
 	configNames := make([]string, 0)
 	// Check all are valid before we delete anything
-	for k, v := range networkConfig.ConfigurationChanges {
-		configChangeIds := manager.GetManager().ConfigStore.Store[k].Changes
-		if store.B64(configChangeIds[len(configChangeIds)-1]) != store.B64(v) {
+	for configName, changeID := range networkConfig.ConfigurationChanges {
+		configChangeIds := manager.GetManager().ConfigStore.Store[configName].Changes
+		if store.B64(configChangeIds[len(configChangeIds)-1]) != store.B64(changeID) {
 			return nil, fmt.Errorf(
 				"the last change on %s is not %s as expected. Was %s",
-				k, v, store.B64(configChangeIds[len(configChangeIds)-1]))
+				configName, changeID, store.B64(configChangeIds[len(configChangeIds)-1]))
+		}
+		err := manager.GetManager().RollbackTargetConfig(string(configName))
+		if err != nil {
+			return nil, err
 		}
```

Comment:
maybe log here the list of targets and their changes - if we could fit them in to the response that would be better (although would require proto change - can be done later)

Reply:
done, I've added them to the message of the response for now